src/makeInfinityAdder.js

```diff
@@ -6,7 +6,33 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (a) => {
+    if (a === undefined) {
+      const res = sum;
+
+      sum = 0;
+
+      return res;
+    }
+
+    sum += a;
+
+    return (b) => {
+      if (b === undefined) {
+        const res = sum;
+
+        sum = 0;
+
+        return res;
+      }
+
+      return adder(b);
+    };
```

Comment:
The second function looks redundant, please check if we can simplify the solution